src/components/ec-file-dropzone/ec-file-dropzone.vue

```diff
@@ -42,82 +42,86 @@
   </div>
 </template>
 
-<script>
+<script setup>
+import {
+  computed,
+  onBeforeUnmount,
+  onMounted,
+  ref,
+} from 'vue';
+
 import EcUploadCloud from '../../images/ec-upload-cloud.vue';
 
-const isNotFolder = file => !!file.type && !!file.size;
+const emit = defineEmits(['change']);
 
-export default {
-  name: 'EcFileDropzone',
-  components: {
-    EcUploadCloud,
-  },
-  props: {
-    isDisabled: {
-      type: Boolean,
-      default: false,
-    },
-  },
-  emits: ['change'],
-  data() {
-    return {
-      dragEnterLeaveEventCounter: 0,
-    };
-  },
-  computed: {
-    isDragging() {
-      return this.dragEnterLeaveEventCounter > 0;
-    },
-  },
-  mounted() {
-    document.addEventListener('dragenter', this.onDragenter);
-    document.addEventListener('dragover', this.onDragover);
-    document.addEventListener('dragleave', this.onDragleave);
-    document.addEventListener('drop', this.onDrop);
-  },
-  beforeUnmount() {
-    document.removeEventListener('dragenter', this.onDragenter);
-    document.removeEventListener('dragover', this.onDragover);
-    document.removeEventListener('dragleave', this.onDragleave);
-    document.removeEventListener('drop', this.onDrop);
+const props = defineProps({
+  isDisabled: {
+    type: Boolean,
+    default: false,
   },
-  methods: {
-    onFileInputChange() {
-      const files = [...this.$refs.fileInput.files].filter(isNotFolder);
-      if (files.length) {
-        this.$emit('change', files);
-        this.$refs.fileInput.value = '';
-      }
-    },
-    onComponentDrop(dragEvent) {
-      if (!this.isDisabled) {
-        const files = [...dragEvent.dataTransfer.files].filter(isNotFolder);
-        if (files.length) {
-          this.$emit('change', files);
-        }
-      }
-      this.disableDragging();
-    },
-    onDrop(dragEvent) {
-      dragEvent.preventDefault();
-      this.disableDragging();
-    },
-    onDragenter() {
-      this.dragEnterLeaveEventCounter++;
-    },
-    onDragover(dragEvent) {
-      // Necessary for the drop listener to work:
-      // https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API#define_a_drop_zone
-      dragEvent.preventDefault();
-    },
-    onDragleave() {
-      this.dragEnterLeaveEventCounter--;
-    },
-    disableDragging() {
-      this.dragEnterLeaveEventCounter = 0;
-    },
-  },
-};
+});
+
+const dragEnterLeaveEventCounter = ref(0);
```

Comment:
Why we use here a number and not a boolean? We could even delete this variable and replace it with

const isDragging = ref(false)

Reply:
The goal of the component is for when a user starts dragging a file anywhere on the page, the dropzone highlights, directing the user to drop the file on it. Because of this, both the `onDragEnter` and `onDragLeave` functions are constantly firing as the user drags their file around, with a boolean it creates unpredictable results. The counter is always in a positive range when a dragging event is occurring and then setting it to zero or when the user leaves the page goes below 0, removing the highlighting.

Reply:
Thank you! I see, sorry.